elastalert/test_rule.py

```diff
@@ -23,63 +36,244 @@ def print_terms(terms, parent):
             print_terms(terms[term], parent + term + '.')
 
 
-def check_files():
-    print("Note: This tool is for testing filters and config syntax. It will not process data or alert.\n")
-    parser = argparse.ArgumentParser(description='Validate a rule configuration')
-    parser.add_argument('files', metavar='file', type=str, nargs='+', help='rule configuration filename')
-    parser.add_argument('--schema-only', action='store_true', help='Show only schema errors; do not run query')
-    parser.add_argument('--days', type=int, default=[1, 7], nargs='+', help='Query the previous N days with this rule')
-    args = parser.parse_args()
+def check_files(args):
+    """ Loads a rule config file, performs a query over the last day (args.days), lists available keys
+    and prints the number of results. """
+    filename = args.file
+    with open(filename) as fh:
+        conf = yaml.load(fh)
+    load_options(conf)
+    print("Successfully loaded %s\n" % (conf['name']))
+
+    if args.schema_only:
+        return []
```

Comment:
check_file makes more sense. It used to take a list of files, but now it only checks one at a time.